File: src/test/java/utils/OS.java
package utils;

public class OS {
    public static String OS = System.getProperty("platformName");

    public static boolean isAndroid() {
        if (OS == null) {
            throw new RuntimeException("Platform is not specified! Please run with -DplatformName=android or -DplatformName=ios");
        }
        return OS.equalsIgnoreCase("android");
    }

    public static boolean isIOS() {
        if (OS == null) {
            throw new RuntimeException("Platform is not specified! Please run with -DplatformName=android or -DplatformName=ios");
        }
        return OS.equalsIgnoreCase("ios");
    }

    public static boolean isMobile() {
        return isAndroid() || isIOS();
    }
}

File: src/test/typescript/utils/OS.ts
export class OS {
    public static OS: string | null = process.env.platformName || null;

    public static isAndroid(): boolean {
        if (this.OS === null) {
            throw new Error("Plateforme non spécifiée ! Veuillez exécuter avec platformName=android ou platformName=ios");
        }
        return this.OS.toLowerCase() === "android";
    }

    public static isIOS(): boolean {
        if (this.OS === null) {
            throw new Error("Plateforme non spécifiée ! Veuillez exécuter avec platformName=android ou platformName=ios");
        }
        return this.OS.toLowerCase() === "ios";
    }

    public static isMobile(): boolean {
        return this.isAndroid() || this.isIOS();
    }
}

